# Video Game Industry Study (2000–2013)

Author: T. Vorobyeva
Date: May, 2025

# Project Goals and Objectives

<font color='#777778'> 
- Conduct a study of the video game industry from 2000 to 2013, focusing on the RPG (Role-playing games) segment across major platforms, game genres, and received scores.<br> 
- The client - game developing company - plans to use these insights to evaluate multiple hypotheses, specifically exploring how game ratings impact overall ROI.</font>

# Data Description

<font color='#777778'>The study utilized several primary data sources: information on game sales, as well as user and critic scores.</font>

# Project Scope

<font color='#777778'>
The research was conducted across three distinct phases, with a rigorous focus on minimizing outliers and data distortions to ensure a high-quality foundation for the final insights:<br>
- Data Preparation (ETL Framework): Implementing comprehensive extraction, validation, cleaning, and transformation processes.<br>
- Data Processing: Structuring and aggregating the refined dataset.<br>
- Data Analysis & Synthesis: Conducting the core research and testing hypotheses.
</font>

# Data Loading and Exploration

In [1]:
# Download the necessary Python libraries and the dataset '/datasets/new_games.csv'.
import pandas as pd
try:
    # look for file in the local folder
    df = pd.read_csv('new_games.csv')
except Exception as e:
    # Look for file on the server
    df = pd.read_csv('https://code.s3.yandex.net/datasets/new_games.csv')

# Show all columns
pd.set_option('display.max_columns', None)

# Show all rows 
pd.set_option('display.max_rows', 20)


In [2]:
# Display the result of the info() method on the screen.
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   Name             16954 non-null  str    
 1   Platform         16956 non-null  str    
 2   Year of Release  16681 non-null  float64
 3   Genre            16954 non-null  str    
 4   NA sales         16956 non-null  float64
 5   EU sales         16956 non-null  str    
 6   JP sales         16956 non-null  str    
 7   Other sales      16956 non-null  float64
 8   Critic Score     8242 non-null   float64
 9   User Score       10152 non-null  str    
 10  Rating           10085 non-null  str    
dtypes: float64(4), str(7)
memory usage: 1.4 MB


# Data Types for Each Column

As seen in the table above, the dataset contains **16,956 records and 11 columns**, providing information about video games, their sales across world regions, and various ratings.

Specifically, it contains the following data: 
• Name — game name
• Platform — platform name
• Year of Release — the year the game was released
• Genre — game genre
• NA sales — sales in North America (in millions of copies sold)
• EU sales — sales in Europe (in millions of copies sold)
• JP sales — sales in Japan (in millions of copies sold)
• Other sales — sales in other countries (in millions of copies sold)
• Critic Score — critic score (from 0 to 100)
• User Score — user score (from 0 to 10)
• Rating — ESRB rating (Entertainment Software Rating Board, which determines computer game ratings and assigns an appropriate age category)

 • **Incorrect data types for certain columns**
Furthermore, some data columns have a string format: "EU sales", "JP sales", "User Score", and "Rating" instead of numerical formats (such as float64), which are preferred for subsequent calculations. "Year of Release" is saved as a float64 instead of a more efficient and logical integer.

 • **Presence of missing values in several columns**
The highest number of missing values is found in the Critic Score, User Score, and Rating columns, which will require a more detailed analysis and potentially alternative workarounds since these data are highly relevant for further analysis. At the same time, a small fraction of missing values in the "Year of Release" column will naturally be filtered out once we slice the dataset by the target time period.



In [3]:
# Display the first few rows of the dataframe on the screen.
df.head() 


,Name,Platform,Year of Release,Genre,NA sales,EU sales,JP sales,Other sales,Critic Score,User Score,Rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN



As the table above shows, the dataframe is sorted by game sales volume in the historically leading North-American market. The top 5 is dominated by games developed for the legendary home console by Nintendo (WII, NES platforms), and only Game Boy (Tetris, Pokemon), which has also been on the market for decades, managed to compete with them in the top 5.


In [4]:
# Request additional information describing the quantitative data. 
display(df.describe())

,Year of Release,NA sales,Other sales,Critic Score
count,16681.000000,16956.000000,16956.000000,8242.000000
mean,2006.485522,0.262023,0.047087,68.926717
std,5.873102,0.808654,0.185577,13.944565
min,1980.000000,0.000000,0.000000,13.000000
25%,2003.000000,0.000000,0.000000,60.000000
50%,2007.000000,0.080000,0.010000,71.000000
75%,2010.000000,0.240000,0.030000,79.000000
max,2016.000000,41.360000,10.570000,98.000000


- As seen from the table above, the database contains games **released from 1980 to 2016.**
- At least 25% of the games in the database belong to the Free-to-Play segment, meaning they generate revenue through in-game microtransactions rather than copy sales. These games received an average critic score of 60%. 
- Meanwhile, 50% of the games were sold in quantities not exceeding 80,000 copies. Taking the North American market as an example, the spread in sales figures is enormous. 
- Based on this and other indicators, an interim conclusion can be drawn that the market is highly fragmented: on one end, there are massive studios with multi-million dollar development and marketing budgets, yielding corresponding revenues. 
- On the other end, a large number of "indie" developers operate with teams of just a few people and settle for "scraps from the master's table" (although there are exceptions like Minecraft).


---

# Assumptions Made


**Data and Methodology Limitations**

During our research, we encountered the following data access constraints:

- **Lack of Financial Metrics and In-Game Revenue**: Market volume is estimated solely by the number of copies sold. This skews the overall picture, particularly for the free-to-play segment.

- **Varying Release Dates**: The dataset only tracks sales from the moment of a game's latest release. Consequently, older titles (e.g., released in 2000) accumulate more sales by the end of the target period (2013) than newer releases (e.g., from 2010).

- **Implication**: These factors may artificially inflate the performance of premium games and the North American market (which historically launched earlier), even though the industry is currently driven by the mobile sector, handheld consoles, and Asia.

**Handling Missing Data**

Since this study is marketing-oriented rather than strategic, the following approach was applied to handle minor data discrepancies:

- Rows with isolated missing values (less than 1%) will be removed.

- Missing values in columns used for calculations will be imputed with mean values.

---

# Data Error Check and Preprocessing

## Dataframe Column Names / Labels

- Display the names of all dataframe columns and verify their naming style.
- Convert all columns to snake_case: new column names will be in lowercase, with spaces replaced by underscores.

In [5]:
df.columns

Index(['Name', 'Platform', 'Year of Release', 'Genre', 'NA sales', 'EU sales',
       'JP sales', 'Other sales', 'Critic Score', 'User Score', 'Rating'],
      dtype='str')

In [6]:
df.columns = df.columns.str.lower().str.replace(' ','_')
print(df.columns) 

Index(['name', 'platform', 'year_of_release', 'genre', 'na_sales', 'eu_sales',
       'jp_sales', 'other_sales', 'critic_score', 'user_score', 'rating'],
      dtype='str')


In [7]:
display(df.head())

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
0,Wii Sports,Wii,2006.0,Sports,41.36,28.96,3.77,8.45,76.0,8,E
1,Super Mario Bros.,NES,1985.0,Platform,29.08,3.58,6.81,0.77,NaN,NaN,NaN
2,Mario Kart Wii,Wii,2008.0,Racing,15.68,12.76,3.79,3.29,82.0,8.3,E
3,Wii Sports Resort,Wii,2009.0,Sports,15.61,10.93,3.28,2.95,80.0,8,E
4,Pokemon Red/Pokemon Blue,GB,1996.0,Role-Playing,11.27,8.89,10.22,1.00,NaN,NaN,NaN


# Data Types

Some numerical data (EU sales, JP sales, User Score) are recorded in string format instead of float64.
Possible reasons:

* **Use of a comma** instead of a period as a decimal separator, or the presence of currency symbols (e.g., €).
* **Missing values** like "no data" or "n/a".
Below, we will convert these columns to the required data type.

In [8]:
# Convert the columns "eu_sales" and "jp_sales" to numeric format and check the result.
df['eu_sales'] = pd.to_numeric(df['eu_sales'], errors='coerce')
df['jp_sales'] = pd.to_numeric(df['jp_sales'], errors='coerce')
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  str    
 1   platform         16956 non-null  str    
 2   year_of_release  16681 non-null  float64
 3   genre            16954 non-null  str    
 4   na_sales         16956 non-null  float64
 5   eu_sales         16950 non-null  float64
 6   jp_sales         16952 non-null  float64
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   float64
 9   user_score       10152 non-null  str    
 10  rating           10085 non-null  str    
dtypes: float64(6), str(5)
memory usage: 1.4 MB


In [9]:
# Selectively check what changed (using eu_sales as an example):
# Look for rows that became missing values (NaN) after conversion.
display(df[df['eu_sales'].isna()])

,name,platform,year_of_release,genre,na_sales,eu_sales,jp_sales,other_sales,critic_score,user_score,rating
446,Rhythm Heaven,DS,2008.0,Misc,0.55,NaN,1.93,0.13,83.0,9,E
802,Dead Rising,X360,2006.0,Action,1.16,NaN,0.08,0.20,85.0,7.6,M
1131,Prince of Persia: Warrior Within,PS2,2004.0,Action,0.54,NaN,0.00,0.22,83.0,8.5,M
1132,Far Cry 4,XOne,2014.0,Shooter,0.80,NaN,0.01,0.14,82.0,7.5,M
1394,Sonic Advance 3,GBA,2004.0,Platform,0.74,NaN,0.08,0.06,79.0,8.4,E
1612,Ratatouille,DS,2007.0,Action,0.49,NaN,0.00,0.14,NaN,NaN,NaN


In [10]:
# Convert the "user_score" column to a numerical format, replacing string values with missing values, and verify the result.
df['user_score'] = pd.to_numeric(df['user_score'], errors='coerce')
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 16956 entries, 0 to 16955
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             16954 non-null  str    
 1   platform         16956 non-null  str    
 2   year_of_release  16681 non-null  float64
 3   genre            16954 non-null  str    
 4   na_sales         16956 non-null  float64
 5   eu_sales         16950 non-null  float64
 6   jp_sales         16952 non-null  float64
 7   other_sales      16956 non-null  float64
 8   critic_score     8242 non-null   float64
 9   user_score       7688 non-null   float64
 10  rating           10085 non-null  str    
dtypes: float64(7), str(4)
memory usage: 1.4 MB


---

## Presence of missing values in the data

In [11]:
# Calculate the number of missing values in each column in absolute and relative terms.
empty_num = df.isna().sum()
empty_share = (empty_num * 100 / df.shape[0]).round(4) 
missing_data = pd.DataFrame({
    'Missing values (units)': empty_num,
    'Share missing values (%)': empty_share
})
display(missing_data)


,Missing values (units),Share missing values (%)
name,2,0.0118
platform,0,0.0000
year_of_release,275,1.6218
genre,2,0.0118
na_sales,0,0.0000
eu_sales,6,0.0354
jp_sales,4,0.0236
other_sales,0,0.0000
critic_score,8714,51.3918
user_score,9268,54.6591


The highest number of missing values is in the "critic_score", "user_score", and "rating" columns. These are likely systematic gaps due to historical or technical reasons:

* **Pre-internet era:** The game was released before the rise of review websites (early 2000s).
* **Niche games:** The game was too obscure to be reviewed or didn't meet the minimum user rating threshold.
* **ESRB fees:** The ESRB rating requires a fee. If a game was only released in Japan or Europe, developers had no reason to pay for it.
* **Technical errors:** Data could have been lost during scraping, collection, or uploading.

<div class="alert alert-info">
<h2> Missing Value Handling Notes <a class="tocSkip"> </h2>

Missing values were handled to enable correct data types and calculations in Python: <br>
    - <b>rating:</b> Replaced NaN with "unknown". <br>
    - <b>name, year_of_release, genre:</b> Dropped rows with missing essential data. <br>
    - <b>eu_sales, jp_sales:</b> Filled missing values with the mean grouped by year and platform. <br>
    - <b>year_of_release:</b> Converted data type to int64.
</div>

In [12]:
# In the "rating" column containing string values, replace NaN with "Unknown".
df['rating'] = df['rating'].fillna('Unknown')

In [16]:
# Drop rows with missing essential information (game name, release year, genre) in columns that will be required for calculations.
df = df.dropna(subset=['name', 'genre', 'year_of_release'])

# Change the data format in the "year_of_release" column to int64.
df['year_of_release'] = df['year_of_release'].astype('int64')

In [15]:
# For eu_sales and jp_sales, fill missing values with the mean grouped by year of release and platform.
mean_eu_sales = df[(df['year_of_release'] >= 2000) & (df['year_of_release'] <= 2013)].groupby(['year_of_release', 'platform'])['eu_sales'].mean()
mean_jp_sales = df[(df['year_of_release'] >= 2000) & (df['year_of_release'] <= 2013)].groupby(['year_of_release', 'platform'])['jp_sales'].mean()
eu_sales_table = pd.DataFrame(mean_eu_sales).reset_index()
jp_sales_table = pd.DataFrame(mean_jp_sales).reset_index()

print(f'Average sales volume per video game in Europe by release year and platform (units):')
display(eu_sales_table.sort_values(by='year_of_release', ascending=False).head(3))
print(f'Average sales volume per video game in Japan by release year and platform (pcs.)')
display(jp_sales_table.sort_values(by='year_of_release', ascending=False).head(3))

Average sales volume per video game in Europe by release year and platform (units):


,year_of_release,platform,eu_sales
114,2013,PS4,0.690625
112,2013,PC,0.191538
110,2013,3DS,0.167849


Average sales volume per video game in Japan by release year and platform (pcs.)


,year_of_release,platform,jp_sales
114,2013,PS4,0.058750
112,2013,PC,0.000000
110,2013,3DS,0.256559


<div class="alert alert-info">
<b> Missing values in the eu_sales and jp_sales columns were filled with the mean values grouped by release year and platform.<br>

Since filtering the dataframe by year was not yet required at this stage of the study, we applied it solely within the scope of this task (filling missing values) to make the results for the period of interest more illustrative.</b> 


---

# 3. Explicit and implicit duplicates in data

# Implicit duplicates in data


In [18]:
# Output sorted unique values in the "genre" column.
unique_genre = sorted(df['genre'].unique())
print(unique_genre)


['ACTION', 'ADVENTURE', 'Action', 'Adventure', 'FIGHTING', 'Fighting', 'MISC', 'Misc', 'PLATFORM', 'PUZZLE', 'Platform', 'Puzzle', 'RACING', 'ROLE-PLAYING', 'Racing', 'Role-Playing', 'SHOOTER', 'SIMULATION', 'SPORTS', 'STRATEGY', 'Shooter', 'Simulation', 'Sports', 'Strategy']


In [20]:
# Intermediate Conclusion: The number of genres is excessively large due to inconsistent text casing. Additionally, for a more streamlined analysis, it may be beneficial to group these genres into broader, consolidated categories.

# Convert the genre names to lowercase and sort them for better visual clarity.

df['genre'] = sorted(df['genre'].str.lower())
print(df['genre'].unique())

<StringArray>
[      'action',    'adventure',     'fighting',         'misc',
     'platform',       'puzzle',       'racing', 'role-playing',
      'shooter',   'simulation',       'sports',     'strategy']
Length: 12, dtype: str


In [21]:
# Print sorted list of unique values in the 'platform' column.
unique_platform = sorted(df['platform'].unique())
print(unique_platform)


['2600', '3DO', '3DS', 'DC', 'DS', 'GB', 'GBA', 'GC', 'GEN', 'GG', 'N64', 'NES', 'NG', 'PC', 'PCFX', 'PS', 'PS2', 'PS3', 'PS4', 'PSP', 'PSV', 'SAT', 'SCD', 'SNES', 'TG16', 'WS', 'Wii', 'WiiU', 'X360', 'XB', 'XOne']


In [22]:
# Platform names need to be converted to uppercase and subsequently — at the analysis stage — grouped into broader categories (e.g., all generations of the same platform) for more convenient analysis.
df['platform'] = df['platform'].str.upper()

In [23]:
# Convert game names to lowercase and remove whitespace from both ends.
df['name'] = df['name'].str.lower().str.strip()

In [24]:
# Output the sorted list of unique values in the 'year_of_release' column
# unique_year = sorted(df['year_of_release'].unique())
unique_year = df['year_of_release'].sort_values().unique().tolist()
print(unique_year)

[1980, 1981, 1982, 1983, 1984, 1985, 1986, 1987, 1988, 1989, 1990, 1991, 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999, 2000, 2001, 2002, 2003, 2004, 2005, 2006, 2007, 2008, 2009, 2010, 2011, 2012, 2013, 2014, 2015, 2016]


Interim conclusion by year: no adjustments needed.

In [25]:
# Print unique values in the "rating" column.
unique_rating = df['rating'].unique().tolist()
print(unique_rating)

['E', 'Unknown', 'M', 'T', 'E10+', 'K-A', 'AO', 'EC', 'RP']


<div class="alert alert-info">
<h2> Interim Conclusion for the "rating" column <a class="tocSkip"> </h2>

As can be seen, it contains several values that do not belong to the ESRB rating system.<br>
 
In reality, there are six main ratings: "E" (Everyone), "E10+" (Everyone 10 and Up), "T" (Teen), "M" (Mature), "AO" (Adults Only), and "RP" (Rating pending). Until 2018, the "EC" (Early childhood) rating was also used.<br>
 
Since we might group by rating in the future, it is necessary to standardize these erroneous values. The proposed solution is as follows:<br>
- Replace "K-A" (kids-adults) values with "E" (Everyone)<br>
- Replace "EC" value with "E".</b> 

In [27]:
# Checking the "rating" column value distribution to decide how to handle erroneous values.
count_rating = df['rating'].value_counts()
share = round(count_rating * 100 / df['rating'].count(), 2)
rating_overview_df = pd.DataFrame({
    'Кол-во оценок': count_rating,
    'Доля (%)': share
})
display(rating_overview_df.sort_index(ascending=True))

,Кол-во оценок,Доля (%)
rating,,
AO,1,0.01
E,3968,23.79
E10+,1414,8.48
EC,8,0.05
K-A,3,0.02
M,1558,9.34
RP,1,0.01
T,2948,17.67
Unknown,6778,40.64


In [28]:
# Replace wrongly written rating values:
# - "K-A" (kids-adults) value with "E" (Everyone)
# - "EC" value with "E"

df['rating'] = df['rating'].str.replace('K-A', 'E')
df['rating'] = df['rating'].str.replace('EC', 'E')


<div class="alert alert-info">
<h2> Interim Conclusion on the Distribution of ESRB Ratings <a class="tocSkip"> </h2>
 
- The largest share in the dataset is occupied by games with an unspecified rating ("Unknown" - 40.64%). This might be because the ESRB primarily rates games for the North American market, or due to a lack of data for older games.<br> 
- In order not to lose this huge chunk of data in future groupings, we will keep the "Unknown" category as a separate group.<br>
 
- Among the known ratings, the E ("Everyone") category leads, covering 23.86% of the total number of games.<br> In second place is the T ("Teen") rating with a share of 17.67%.

## Explicit duplicates in the data


In [29]:
# Check the number of unique values per column and their percentage of total rows (and round the results).
unique_count = df.nunique() 
share_unique_count = (unique_count * 100 / df.shape[0]).round(2)
unique_values = pd.DataFrame({'unique_values_count': unique_count,
                        'share': share_unique_count})
print(f'Total rows in the dataframe: {df.shape[0]}')
print(f'Unique value count and share by column (в %):')
display(unique_values)

Total rows in the dataframe: 16679
Unique value count and share by column (в %):


,unique_values_count,share
name,11426,68.51
platform,31,0.19
year_of_release,37,0.22
genre,12,0.07
na_sales,401,2.40
eu_sales,307,1.84
jp_sales,244,1.46
other_sales,155,0.93
critic_score,81,0.49
user_score,95,0.57


In [30]:
# Checking for explicit duplicates.
full_duplicates = df.duplicated().sum()
print(f'The number of exact duplicates in the table: {full_duplicates}')

The number of exact duplicates in the table: 235


In [31]:
# Remove the duplicate rows found and reset the dataframe index (to avoid gaps).
df = df.drop_duplicates()
df = df.reset_index(drop=True)

In [32]:
# Calculate the absolute and relative number of deleted rows."
initial_row_num = 16956
new_row_num = df.shape[0]
deleted_num = initial_row_num - new_row_num
share_deleted = round((deleted_num * 100 / initial_row_num), 2)
unique_count_names = df['name'].nunique() 
print(f'The number of deleted rows: {deleted_num}') 
print(f'The share of deleted rows: {share_deleted}%') 
print(f'The new number of rows in the dataframe: {new_row_num}') 
print(f'The unique number of games: {unique_count_names}') 

The number of deleted rows: 512
The share of deleted rows: 3.02%
The new number of rows in the dataframe: 16444
The unique number of games: 11426


# Data Preprocessing Summary


After cleaning, the dataset contains **16,432 rows** and **11,421 unique titles**.<br> This gap is standard for the gaming industry, where a single game can be released across multiple platforms; thus, the rows are valid and preserved for analysis.<br>

Preprocessing successfully normalized key metrics:
* **Genres**: reduced from 24 to 12.
* **Platforms**: the total count (31) aligns with industry standards.
* **Scores**: critic and user ratings now show a smooth, standard distribution (on 100-point and 10-point scales, respectively).


# Data Filtering 

The stakeholders want to study historical game sales data from the early 21st century, specifically focusing on the period between 2000 and 2013, inclusive.

In [33]:
# Prepare a data slice for 2000-2013 inclusive in a new dataframe df_actual
df_actual = df[(df['year_of_release'] >= 2000) & (df['year_of_release'] <= 2013)].copy()
df_actual.info()

# Check the results
start_year = df_actual['year_of_release'].min()
end_year = df_actual['year_of_release'].max()
print(f'First year of the analyzed period:')
print(start_year)
print(f'Last year of the analyzed period:')
print(end_year)

<class 'pandas.DataFrame'>
Index: 12781 entries, 0 to 16442
Data columns (total 11 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   name             12781 non-null  str    
 1   platform         12781 non-null  str    
 2   year_of_release  12781 non-null  int64  
 3   genre            12781 non-null  str    
 4   na_sales         12781 non-null  float64
 5   eu_sales         12776 non-null  float64
 6   jp_sales         12777 non-null  float64
 7   other_sales      12781 non-null  float64
 8   critic_score     7169 non-null   float64
 9   user_score       6483 non-null   float64
 10  rating           12781 non-null  str    
dtypes: float64(6), int64(1), str(4)
memory usage: 1.2 MB
First year of the analyzed period:
2000
Last year of the analyzed period:
2013


---

# Data Categorization
    
Let's categorize the data: We will group all games by their "user_score" into the following categories: "high score" (from 8 to 10 inclusive), "medium score" (from 3 up to but not including 8), and "low score" (from 0 up to but not including 3). 


In [34]:
df_actual['user_score_group'] = pd.cut(df_actual['user_score'], bins=[-0.1, 3, 8, 11], labels=["low score", "middle score", "high score"], right=False)
# Display the first 10 rows for verification.
display(df_actual[['name', 'user_score', 'user_score_group']].head(10))

,name,user_score,user_score_group
0,wii sports,8.0,high score
2,mario kart wii,8.3,high score
3,wii sports resort,8.0,high score
6,new super mario bros.,8.5,high score
7,wii play,6.6,middle score
8,new super mario bros. wii,8.4,high score
10,nintendogs,NaN,NaN
11,mario kart ds,8.6,high score
13,wii fit,7.7,middle score
14,kinect adventures!,6.3,middle score


- Divide all games by critic scores ("critic_score") and distinguish the following categories: "high score" (from 80 to 101 inclusive), "medium score" (from 30 to 80, excluding the right boundary of the interval), and "low score" (from 0 to 30, excluding the right boundary of the interval).

In [35]:
df_actual['critic_score_group']  = pd.cut(df_actual['critic_score'], bins=[-0.1, 30, 80, 101], labels=["low score", "middle score", "high score"], right=False)

# Display the first 10 rows for verification.
display(df_actual[['name', 'critic_score', 'critic_score_group']].head(10))

,name,critic_score,critic_score_group
0,wii sports,76.0,middle score
2,mario kart wii,82.0,high score
3,wii sports resort,80.0,high score
6,new super mario bros.,89.0,high score
7,wii play,58.0,middle score
8,new super mario bros. wii,87.0,high score
10,nintendogs,NaN,NaN
11,mario kart ds,91.0,high score
13,wii fit,80.0,high score
14,kinect adventures!,61.0,middle score


- After categorizing the data, let's verify the result: group the data by the selected categories and count the number of games in each category.

In [36]:
# "Calculating dataframe distribution by user rating categories and creating an output table.
count_user_val = df_actual['user_score_group'].value_counts()
share_user_val = (df_actual['user_score_group'].value_counts(normalize=True) * 100).round(2)

grouped_user_table = pd.DataFrame({
    'Score count': count_user_val,
    'Share (%)': share_user_val
})

display(grouped_user_table.sort_index())


,Score count,Share (%)
user_score_group,,
low score,116,1.79
middle score,4081,62.95
high score,2286,35.26


In [37]:
# Calculating dataframe distribution by сritic rating categories and creating an output table.

count_critic_val = df_actual['critic_score_group'].value_counts()
share_critic_val = (df_actual['critic_score_group'].value_counts(normalize=True) * 100).round(2)
grouped_critic_table = pd.DataFrame({
    'Score count': count_critic_val,
    'Share (%)': share_critic_val
})

display(grouped_critic_table.sort_index())

,Score count,Share (%)
critic_score_group,,
low score,55,0.77
middle score,5422,75.63
high score,1692,23.60


<div class="alert alert-info">

The vast majority of games (over 98%) received average and high ratings from both users and critics.

## Let’s identify the top 7 platforms by the number of games released over the entire period (2000–2013).

- We will break this task down into subtasks: first, group the data by platform, count the number of games, and finally, sort the results in descending order to find the top 7.

In [38]:
grouped_platform = df_actual['platform'].value_counts()
share_platform = (df_actual['platform'].value_counts(normalize = True)* 100).round(2)
name_platform = grouped_platform.index
platform_table = pd.DataFrame({'Platform title': name_platform, 'Game number': grouped_platform, 'Game share in %': share_platform}, index=grouped_platform.index)
sorted_platform_table = platform_table.sort_values(by= 'Game number', ascending=False)

print(f'The top 7 platforms by the number of games:')
display(sorted_platform_table.head(7))


The top 7 platforms by the number of games:


,Platform title,Game number,Game share in %
platform,,,
PS2,PS2,2127,16.64
DS,DS,2120,16.59
WII,WII,1275,9.98
PSP,PSP,1180,9.23
X360,X360,1121,8.77
PS3,PS3,1087,8.50
GBA,GBA,811,6.35


The top spots in the rating are held with a minimal gap between them by the technologically powerful platforms PS2 (2,127 games) and DS (2,120 games), which together account for more than a third of all games on the market. They are followed by the Wii (1,275 games), PSP (1,126 games), and X360 (1,121 games) platforms, which once took the market by storm with their innovation.

---

# Summary
- **Data Preprocessing**: The data was verified, filtered, and cleaned. The initial dataset was of high quality, requiring the removal of only 3% of the rows.<br><br> 
- **Dataset Specifics**: The final dataset includes 16,432 rows and 11,421 unique game titles. The duplicate titles are due to industry specifics (multi-platform releases), so these rows were kept as valid for further analysis.<br><br>
- **Feature Correction**: Post-cleaning, the number of genres was optimized from 24 to 12, and the total number of platforms stands at 31, aligning with industry standards. Critic and user score distributions appear realistic (on standard 100-point and 10-point scales).<br><br>
- **Score Distribution**: The majority of games received average-to-high ratings from both critics and users.<br><br>
- **Top Platforms by Game Count**: The technologically advanced DS (2,110 games) and PS2 (2,044 games) lead the ranking. They are followed by Wii (1,261) and PSP (1,126), platforms that once disrupted the market with their innovativeness.


# Recommendations for Further Analysis
To ensure statistical integrity, the following methodological adjustments are recommended:

- **Exclude Free-to-Play (F2P) from Core Sales**: Omit F2P titles from primary sales analyses. Because standard metrics miss in-game revenue (microtransactions, battle passes), including them artificially deflates average revenue and distorts financial data.

- **Segment the Market**: For comprehensive views, strictly separate the dataset into Premium (Buy-to-Play) and Free-to-Play (F2P). Analyzing these independently prevents data contamination and yields more accurate insights.

- **Standardize Genre Comparisons via Sentiment**: Evaluate genre popularity using the mean values of User and Critic Scores. This normalizes the data, reflecting actual acclaim independent of sales volume.
